src/sentry/superuser_access/prevent_superuser_access_backend.py

```diff
@@ -0,0 +1,19 @@
+from sentry.models.organization import Organization
+from sentry.organizations.services.organization import RpcUserOrganizationContext
+from sentry.utils.services import Service
+
+
+class PreventSuperuserAccessBackend(Service):
+    def should_allow_superuser_access(
+        self,
+        organization_context: Organization | RpcUserOrganizationContext,
+    ) -> bool:
+        raise NotImplementedError
+
+
+class PreventSuperuserAccess(Service):
+    def should_allow_superuser_access(
```

Comment:
It would be good to have a docstring on why there are two classes. You could skip those docs by having `PreventSuperuserAccess(PreventSuperuserAccessBackend)` which makes the relation between the classes clear.

Having a docstring explaining that this is a saas only component used in getsentry could be helpful for future us as well.